Java-Inheritance.md

```diff
@@ -0,0 +1,172 @@
+# Java Inheritance
+
+Java inheritance refers to the ability of a Java Class to `inherit` the properties from some other Class. Think of it like a child inheriting properties from its parents, the concept is very similar to that. In Java lingo, it is also called _extend_-ing a class. Some simple things to remember :
+
+* The Class that extends or inherits is called **subclass**
+* The Class that is being extended or inherited is called **superclass**
+
+Thus, inheritance gives Java the cool capability of _re-using_ code, or share code between classes! Let's describe it with the classic example of a `Vehicle` class and a `Car` class :
+
```

Comment:
add a heading 2 break point here for our sweet beloved @camperbot to extract upto this point for gitter chat 

Like this,

```md
## (just add a space after the two ##)
```

Move the line `Let's describe it with the classic example of a `Vehicle` class and a `Car` class :` after the heading.